### PR TITLE
Binary log file

### PR DESCRIPTION
**Pull request by juryghidinelli**
The log file without that replacement, is interpreted as binary file log issue.

### CODE REVIEW

**Comment by baudehlo** [comment]
Which log plugin are you using?

**Comment by juryghidinelli** [comment]
The plugin that i use is https://github.com/haraka/haraka-plugin-log-reader

File: plugins/clamd.js [hunk 1]
```diff
@@ -232,7 +232,7 @@ exports.hook_data_post = function (next, connection) {
 
         var result = '';
         socket.on('line', function (line) {
-            connection.logprotocol(plugin, 'C:' + line);
+            connection.logprotocol(plugin, 'C:' + line.split('').filter(function(x) { var n = x.charCodeAt(0); return 31 < n && 127 > n  }).join('') );
```
Comment (msimerson) [inline plugins/clamd.js]: We can't merge b/c lint tests are failing. We require a space after a function declaration. IE, this change:
```diff
-connection.logprotocol(plugin, 'C:' + line.split('').filter(function(x) { var n = x.charCodeAt(0); return 31 < n && 127 > n  }).join('') );
+connection.logprotocol(plugin, 'C:' + line.split('').filter(function (x) { var n = x.charCodeAt(0); return 31 < n && 127 > n  }).join('') );
```

**Comment by baudehlo** [comment]
And I'd rather not merge if it turns out the fix should be in logger.*
functions instead. Any plugin could log a \x00, not just this one.

**Comment by msimerson** [comment]
> And I'd rather not merge if it turns out the fix should be in logger.* functions instead. Any plugin could log a \x00, not just this one.

I'm not so sure that's the best idea. Checking every log entry (and Haraka emits TONS of them) for the presence of binary characters sounds a LOT more expensive than applying the filters in the few locations where binary may be output.

**Comment by baudehlo** [comment]
True. But if it turns out to be in plugin-log-reader that might not be a
bad place to fix it.

**Pull request closed without merging**